### PR TITLE
[MIG] hr: Migrated to new api.

### PR DESCRIPTION
**Pull request by rgo-odoo**
Task: https://www.odoo.com/web#id=12437&view_type=form&model=project.task&action=327
Pad: https://pad.odoo.com/p/r.ECFkv1NxYlPWI7gg

### CODE REVIEW

**Comment by atp-odoo** [comment]
it's difficult to review when file is moved in models and github treat it as New file in PR. 
i think for now better we can keep as it is (don't move in models/views directory) those file which has 60 to 70% changes

**Comment by rgo-odoo** [comment]
@atp-odoo I think reviewing the commit(for the branch having 60%-70% changes) would relatively be easier than reviewing PR in this case. Like, it can be easier to review "hr.py" from the specific commit, https://github.com/odoo-dev/odoo/commit/51103ae270c6804c503c1e9df45d077d5e08ff89#diff-7fe5c1ac38eeb355c75fa18a98b62323L1
What do you say ?

**Comment by atp-odoo** [comment]
@rgo-odoo yes but then i have to make review comment in separate file and i think , i saw that 
this review comment is not seen in conversation of PR! I am not sure for this.

File: addons/hr/models/hr.py [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+
+class HrJob(models.Model):
+    _name = "hr.job"
+    _description = "Job Position"
+    _inherit = ['mail.thread']
+
+    @api.one
+    @api.depends('no_of_recruitment', 'employee_ids')
+    def _compute_employees(self):
+        nb_employees = len(self.employee_ids or [])
```
Comment (atp-odoo) [inline addons/hr/models/hr.py]: no need of "or []" , len(self.employee_ids) is enough

File: addons/hr/models/hr.py [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+    def set_open(self):
+        return self.write({
+            'state': 'open',
+            'no_of_recruitment': 0,
+            'no_of_hired_employee': 0
+        })
+
+    @api.multi
+    def copy_data(self, default=None):
+        if default is None:
```
Comment (atp-odoo) [inline addons/hr/models/hr.py]: why copy_data() method here? can not work with copy() method with @api.one.

File: addons/hr/models/res_users.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+    @api.multi
+    def stop_showing_employees_suggestions(self):
+        """Update display_employees_suggestions value to False"""
+        for user in self:
+            user.display_employees_suggestions = False
+
+    @api.model
+    def _message_post_get_eid(self):
+        rec = self
```
Comment (atp-odoo) [inline addons/hr/models/res_users.py]: why assert removed ? useless decorator here. ensure_one work here

File: addons/hr/models/res_users.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        if 'thread_model' in self.env.context:
+            rec = self.with_context(thread_model='hr.employee')
```
Comment (atp-odoo) [inline addons/hr/models/res_users.py]: rec = self.with_context ? ?

**Comment by rgo-odoo** [comment]
@atp-odoo Changes done.
Thanks for the review.

**Pull request closed without merging**